server/manageuser.php

```diff
@@ -0,0 +1,157 @@
+<?php
+
+require_once 'config.php';
+require_once "user.class.php";
+
+$USER = new User();
+
+if (!$USER->hasPermission('edit_user_permission')) {
+	header("Location: /unauthorized");
+	die();
+}
+
+if(isset($_REQUEST['callback'])){
+	switch ($_REQUEST['callback']) {
+		case 'getUserTable':
+			getUserTable($_REQUEST);
+			break;
+		case 'updateUserPermissions':
+			updateUserPermissions($_REQUEST);
+			break;
+		case 'addUser':
+			addUser($_REQUEST);
+			break;
+		default:
+			die('Unregistered command. This instance has been reported.');
+			break;
+	}
+}
+
+function getUserTable($data){
+	GLOBAL $DB_CONN;
+
+	$response = array();
+	$response['success'] = true;
+
+	$stmt = $DB_CONN->prepare("SELECT userId, firstName, lastName, email, preparesTaxes, archived 
+		FROM vita.user
+		ORDER BY firstName, lastName");
+
+	$stmt->execute(array());
+
+	$thead = "<thead><tr><th>Name</th><th>Email</th><th>Permissions</th><th>Count</th></tr>";
+	$tbody = "<tbody>";
+	while($row = $stmt->fetch(PDO::FETCH_ASSOC)){
+		$optionList = getUserPermissionOptionList($row['userId']);
+
+		$tbody.= "<tr data-user-id='".$row['userId']."'>";
+		$tbody.= "<td>".$row['firstName']." ".$row['lastName']."</td>";
+		$tbody.= "<td>".$row['email']."</td>";
+		$tbody.= "<td><select class='userPermissionList selectpicker' multiple=true>".implode('', $optionList['options'])."</select></td>";
+		$tbody.= "<td>".$optionList['hasPermissionCount']."</td>";
+		$tbody.= "</tr>";
+	}
+	$tbody.= "</tbody>";
+	$thead.= "</tr></thead>";
+
+	$response['table'] = $thead.$tbody;
+
+	print json_encode($response);
+}
+
+// Returns string of html <option> elments with all permissions
+function getUserPermissionOptionList($userId){
+	GLOBAL $DB_CONN;
+
+	$stmt = $DB_CONN->prepare("SELECT 
+		permissionId,
+		name,
+        description,
+        lookupName,
+        (SELECT userPermissionId FROM UserPermission WHERE userId = ? AND permissionId = p.permissionId) as userPermissionId
+	FROM Permission p");
+
+	$stmt->execute(array($userId));
+
+	$options = array();
+
+	$count = 0;
+	while($row = $stmt->fetch(PDO::FETCH_ASSOC)){
+		if($row['userPermissionId']){
+			$selected = 'selected=true';
+			$data = "data-userPermissionId='".$row['userPermissionId']."'";
+			$count++;
+		}else{
+			$selected = '';
+			$data = '';
+		}
+		$options[] = "<option $data value=".$row['permissionId']." $selected>".$row['name']."</option>";
+	}
+
+	return array(
+		'options' => $options,
+		'hasPermissionCount' => $count
+	);
+}
+
+function updateUserPermissions($data){
+	GLOBAL $DB_CONN;
+	GLOBAL $USER;
+
+
+	$response = array();
+	$response['success'] = true;
+
+	$DB_CONN->beginTransaction();
+
+	if(isset($data['removePermissionArr'])){
+		$stmt = $DB_CONN->prepare("DELETE FROM UserPermission WHERE userPermissionId = ?");
+
+		foreach ($data['removePermissionArr'] as $userPermissionId) {
```

Comment:
I'm not sure how I feel about this part. It would be better to do a bulk delete / insert rather than foreach-ing over all of them. But for right now we probably don't need to worry about it. 
The solution that comes to mind is basically adding the proper number of ? and then inserting an array. But I don't anticipate this permission changing handling extremely high volumes that it's necessary. So totally up to you